src/field.rs

```diff
@@ -420,6 +421,12 @@ macro_rules! make_field {
             }
         }
 
+        impl ConstantTimeEq for $elem {
```

Comment:
This `impl` is available anywhere one is working with a concrete type that implements `FieldElement` via this macro. Should we add `ConstantTimeEq` to `trait FieldElement`'s bounds so that it's available when working generically with `impl FieldElement`?